code/menuui/mainhallmenu.cpp

```diff
@@ -1485,6 +1467,9 @@ void main_hall_mouse_release_region(int region)
 		}
 
 		auto sound = Main_hall->door_sounds.at(region).at(1);
+		if (Vasudan_funny && (main_hall_is_retail_vasudan() || main_hall_allows_headz())) {
```

Comment:
Oh very neat PR! Just looking over it, and I wanted to check about something. This section looks like it overrides the sound for all regions, whereas the standard cheat overrides the sound only for the `OPTIONS_REGION` (original lines 543-565 in manhallmenu.cpp). 

Just wanted to check with you to see why that was not changed in `main_hall_set_door_headz`? [as within the `main_hall_set_door_headz` function it the animation and background are changed just for the one region]. Thanks!

Reply:
Oh good catch. I had to dive back into the code to remember what I was doing even. `main_hall_set_door_headz` works for the anim because of a quirk in how we actually handle bitmaps with bmpman. Basically it replaces the currently loaded anim with the headz one and the rest of the code doesn't really care; it just plays whatever animation is in that slot. We can't do that for sounds. That's why this code is here and not in `main_hall_set_door_headz`.

But you're right this needs a block that checks that we're playing a sound only for the headz region so I've added that.